design1.py

```diff
@@ -0,0 +1,82 @@
+
+# 1.use two stacks
+# 2.one stack to store values and one to store min values
+# 3.use global minimum values to keep tarck.
+
+# push,pop,top and get_min time complexity-O(1)
+
+
+class MinStack:
+
+    def __init__(self):
+        
+        self.instack  = [ ]
+        self.minstack = [ ]
+        
+        self.minvalue= float('inf')
+        self.minstack.append(self.minvalue)
+        
+    def push(self, x: int) -> None:
+        
+        self.instack.append(x)
+        self.minvalue=min(self.minvalue,x)
+        self.minstack.append(self.minvalue)
+        
+
+    def pop(self) -> None:
+        self.minstack.pop()
+        self.minvalue=self.minstack[-1]
+        return self.instack.pop()
+
+    def top(self) -> int:
+        return self.instack[-1]
```

Comment:
Please add edge cases, to prevent from pop() peek() on operating on an empty stack.